internal/configs/transportserver_test.go

```diff
@@ -4,6 +4,7 @@ import (
 	"reflect"
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
```

Comment:
I mentioned previously about the pointer override, assuming this is not intended behaviour, can we add a test to confirm that two upstreams generates two health checks ? I think there's multiple health checks in the tests below, but some of them are invalid ?